pkg/mover/chart.go

```diff
@@ -236,17 +276,127 @@ func namespacedPath(fullpath, chartName string) string {
 }
 
 /*
-Move executes the Chart relocation which includes
+  Move perfomes the relocation.
+
+A regular move executes the Chart relocation which includes
 
 1 - Push all the images to their new locations
 
 2 - Rewrite the Helm Chart and its subcharts
 
 3 - Repackage the Helm chart as toChartFilename
+
+A move to tar will:
+
+1 - Drop all images locally
+
+2 - Pack also the original chart (unpacked) and the hints file
+
+3 - Package all in a single compressed tarball
 */
 func (cm *ChartMover) Move() error {
+	if cm.targetOfflineTar != "" {
+		return cm.archive()
+	}
+	return cm.moveChart()
+}
+
+func (cm *ChartMover) archive() error {
 	log := cm.logger
+	log.Printf("Archiving %s@%s...\n", cm.chart.Name(), cm.chart.Metadata.Version)
 
+	tarFolder := cm.targetOfflineTar + ".folder"
+	if err := os.MkdirAll(tarFolder, DefaultTarPermissions); err != nil {
+		return fmt.Errorf("failed to create tar folder %s: %w", tarFolder, err)
+	}
+	if err := archiveChart(tarFolder, cm.chartOrigin); err != nil {
+		return fmt.Errorf("failed archiving chart %s: %w", cm.chart.Name(), err)
+	}
+	cm.logger.Println("Archived char tarball")
+	if err := archiveImages(tarFolder, cm.imageChanges, cm.logger); err != nil {
+		return fmt.Errorf("failed archiving images: %w", err)
+	}
+	if cm.externalHintsFile == "" {
+		return nil
+	}
+	log.Printf("Inserting hints file %s...\n", cm.externalHintsFile)
+	archivedHints := filepath.Join(tarFolder, ArchivedHintsFilename)
+	return copyFile(cm.externalHintsFile, archivedHints)
+}
+
+func archiveChart(folder, chartPath string) error {
+	target := filepath.Join(folder, filepath.Base(chartPath))
+	if err := os.Mkdir(target, DefaultTarPermissions); err != nil {
+		return err
+	}
+	return copyRecursive(chartPath, target)
+}
+
+func copyRecursive(source, destination string) error {
```

Comment:
I should move this to a `files.go` and add independent testing.

If we already use a library that provide this operations I can use it instead. But I would not add a new dependency just for 20 lines of code.